embedded/EscapeROOM/radio_morse_panel/radio_morse_panel.c

```diff
@@ -22,139 +48,84 @@ void unit()
 }
 
-void letter_space()
+void PlayLetterSpace()
 {
-	unit(); unit(); unit();
+	PlayUnit(); PlayUnit(); PlayUnit();
 }
 
-void word_space()
+void PlayWordSpace()
 {
-	unit(); unit(); unit(); unit(); unit(); unit(); unit();
+	PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit(); PlayUnit();
 }
 
-void zero()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void one()
-{
-	dot();  short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void two()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void three()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dash(); short_space();
-	dash();
-}
-
-
-void four()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dash();
-}
-
-
-void five()
-{
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot();
-}
-
-void six()
-{
-	dash(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot();
-}
-
-void seven()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dot(); short_space();
-	dot(); short_space();
-	dot();
+unsigned char GetLetterPositionInMorseAlphabet(char Letter, const MorseCode *MorseAlphabet, const unsigned char NumOfElements){
+	unsigned char LetterPosition;
+	
+	// simply compare each element in alphabet;
+	for(LetterPosition = 0; LetterPosition < NumOfElements; LetterPosition++){
+		if( (MorseAlphabet + LetterPosition)->Symbol == Letter)
+			break;
+	}
+	
+	return LetterPosition;
 }
 
-void eight()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dot(); short_space();
-	dot();
+void PlayMorseLetter(char Letter, const MorseCode *MorseAlphabet, const unsigned char NumOfElements){
+	unsigned char LetterPositionInAlphabet;
+	unsigned char i;
+	
+	LetterPositionInAlphabet = GetLetterPositionInMorseAlphabet(Letter, MorseAlphabet, NumOfElements);
+	
+	// play each morse letter code element 
+	for(i = 0; i < MAX_MORSE_PARTS && (MorseAlphabet + LetterPositionInAlphabet)->Code[i] != '\0'; i++){
+		switch( (MorseAlphabet + LetterPositionInAlphabet)->Code[i] ){
+			case '.' :
+				PlayDot();
+				break;
+			case '-' :
+				PlayDash();
+				break;
+		}
+		PlayShortSpace();
+	}
+	
+	return;
 }
 
-void nine()
-{
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dash(); short_space();
-	dot();
+void PlayMorsePhrase(const char *MorsePhraseToPlay, const MorseCode *MorseAlphabet, const unsigned char NumOfElements){
+		unsigned char i;
+		
+		// play each letter until 'MorsePhraseToPlay' end is not reached
+		for(i = 0; MorsePhraseToPlay[i] != '\0'; i++){
+			PlayMorseLetter(MorsePhraseToPlay[i], MorseAlphabet, NumOfElements);
+			PlayLetterSpace();
+		}
+		
+		PlayWordSpace();
```

Comment:
Maybe add special character (for example " ", hex code 0x20) in morseCode array to allow encoding word space in MorsePhraseToPlay?
```
const char MorsePhraseToPlay[] = "472 ";
```